spanner/spec/spanner_samples_spec.rb

```diff
@@ -0,0 +1,452 @@
+                  instance_id: @instance.instance_id,
+                  database_id: database.database_id
+    }.to output("Inserted data\n").to_stdout
+
+    singers = client.execute("SELECT * FROM Singers").rows.to_a
+    expect(singers.count).to eq 5
+    expect(singers.find {|s| s[:FirstName] == "Catalina" }).not_to be nil
+
+    albums = client.execute("SELECT * FROM Albums").rows.to_a
+    expect(albums.count).to eq 5
+    expect(albums.find {|s| s[:AlbumTitle] == "Go, Go, Go" }).not_to be nil
+  end
+
+  example "query data" do
+    database = create_singers_albums_database
+    client   = @spanner.client @instance.instance_id, database.database_id
+
+    # Insert Singers and Albums (re-use insert_data sample to populate)
+    insert_data project_id:  @project_id,
+                instance_id: @instance.instance_id,
+                database_id: database.database_id
+
+    capture do
+      query_data project_id:  @project_id,
+                 instance_id: @instance.instance_id,
+                 database_id: database.database_id
+    end
+
+    expect(captured_output).to include "1 1 Go, Go, Go"
+    expect(captured_output).to include "1 2 Total Junk"
+    expect(captured_output).to include "2 1 Green"
+    expect(captured_output).to include "2 2 Forever Hold your Peace"
+    expect(captured_output).to include "2 3 Terrified"
+  end
+
+  example "read data" do
+    database = create_singers_albums_database
+    client   = @spanner.client @instance.instance_id, database.database_id
+
+    # Insert Singers and Albums (re-use insert_data sample to populate)
+    insert_data project_id:  @project_id,
+                instance_id: @instance.instance_id,
+                database_id: database.database_id
+
+    capture do
+      read_data project_id:  @project_id,
+                instance_id: @instance.instance_id,
+                database_id: database.database_id
+    end
+
+    expect(captured_output).to include "1 1 Go, Go, Go"
+    expect(captured_output).to include "1 2 Total Junk"
+    expect(captured_output).to include "2 1 Green"
+    expect(captured_output).to include "2 2 Forever Hold your Peace"
+    expect(captured_output).to include "2 3 Terrified"
+  end
+
+  example "create index" do
+    database = create_singers_albums_database
+
+    expect(database.ddl(force: true).join).not_to include(
+      "CREATE INDEX AlbumsByAlbumTitle ON Albums(AlbumTitle)"
+    )
+
+    capture do
+      create_index project_id:  @project_id,
+                   instance_id: @instance.instance_id,
+                   database_id: database.database_id
+    end
+
+    expect(captured_output).to include "Waiting for database update to complete"
+    expect(captured_output).to include "Added the AlbumsByAlbumTitle index"
+
+    expect(database.ddl(force: true).join).to include(
+      "CREATE INDEX AlbumsByAlbumTitle ON Albums(AlbumTitle)"
+    )
+  end
+
+  example "create storing index" do
+    database = create_singers_albums_database
```

Comment:
Database is missing the MarketingBudget column and that's why the expectation on line 190 fails. It passes for me once I added this column. 

```ruby
 add_column project_id:    @project_id,
                      instance_id:  @instance.instance_id,
                      database_id: database.database_id
```